Pod/Classes/PINRemoteImageManager.m

```diff
@@ -1010,14 +1007,19 @@ - (void)didReceiveData:(NSData *)data forTask:(NSURLSessionDataTask *)dataTask
     [self unlock];
     
     [progressiveImage updateProgressiveImageWithData:data expectedNumberOfBytes:[dataTask countOfBytesExpectedToReceive]];
+
     if (hasProgressBlocks && [[self class] isiOS8OrGreater]) {
-        UIImage *progressImage = [progressiveImage currentImage];
-        if (progressImage) {
-            [self lock];
-                task = [self.tasks objectForKey:[self cacheKeyForURL:[[dataTask originalRequest] URL] processorKey:nil]];
-                [task callProgressWithQueue:_callbackQueue withImage:progressImage];
-            [self unlock];
-        }
+        __weak typeof(self) weakSelf = self;
+        [_concurrentOperationQueue pin_addOperationWithQueuePriority:PINRemoteImageManagerPriorityLow block:^{
+            typeof(self) strongSelf = weakSelf;
+            UIImage *progressImage = [progressiveImage currentImage];
+            if (progressImage) {
+                [strongSelf lock];
+                    PINRemoteImageDownloadTask *task = [strongSelf.tasks objectForKey:[strongSelf cacheKeyForURL:[[dataTask originalRequest] URL] processorKey:nil]];
```

Comment:
This could be a little more readable if we pull the key generation out into a local variable:

```objc
NSString *cacheKey = [strongSelf cacheKeyForURL:[[dataTask originalRequest] URL] processorKey:nil];
PINRemoteImageDownloadTask *task = strongSelf.tasks[cacheKey];
```